File: java/src/com/db4o/dg2db4o/chapter10/BackupDemo.java
package com.db4o.dg2db4o.chapter10;

import com.db4o.*;
import java.io.*;

public class BackupDemo {
    
    public static void main(String[] args){
        // set up some logging to demo this feature also
        Db4o.configure().messageLevel(3);
        Db4o.configure().setOut(System.out);
        //Db4o.configure().setOut(new PrintStream(new File("log.txt"));
        
        new File("C:/primary.yap").delete();
        new File("C:/secondary.yap").delete();
        ObjectContainer db = Db4o.openFile("C:/primary.yap");
        db.set(new Person("Stefan", 39));
        
        db.commit();
        try {
            db.ext().backup("C:/secondary.yap");
        } catch (Exception e) {
            e.printStackTrace();
            db.close();
        }
        db.close();
        System.out.println("Done backup");
    }
    
}

File: csharp/Chapter10/BackUpDemo.cs
using System;
using com.db4o;

namespace com.db4o.dg2db4o.chapter10
{
    public class BackUpDemo
    {
        public static void Run()
        {
            Console.WriteLine("\nBACKUP DEMO");
            System.IO.File.Delete("C:/primary.yap");
            System.IO.File.Delete("C:/secondary.yap");
            ObjectContainer db = Db4o.OpenFile("C:/primary.yap");
            db.Set(new Person("Stefan", 39));
            db.Commit();
            try
            {
                db.Ext().Backup("C:/secondary.yap");
            }
            catch (Exception e)
            {
                Console.WriteLine(e.StackTrace);
                db.Close();
            }
            db.Close();

            Console.WriteLine("\nEND BACKUP DEMO");

        }
    }
}
